File: Creational/Singleton/java/SingletonDesignPattern.java
public class SingletonDesignPattern {
    private static class Singleton {
        private static Singleton INSTANCE;

        private Singleton() {
        }

        public static Singleton getSingletonInstance() {
            if (INSTANCE == null) {
                INSTANCE = new Singleton();
            }
            return INSTANCE;
        }

        public void printSingleton() {
            System.out.println("Singleton Class");
        }
    }
    public static void main(String[] args) {
        Singleton singleton = Singleton.getSingletonInstance();
        singleton.printSingleton();
    }

}

File: Creational/Singleton/csharp/SingletonDesignPattern.cs
using System;

public class SingletonDesignPattern
{
    static void Main(String[] args)
    {
        Singleton singleton = Singleton.getSingletonInstance();

        singleton.printSingleton();
    }
}

class Singleton
{
    private static Singleton obj;

    private Singleton()
    {

    }

    public static Singleton getSingletonInstance()
    {
        if (obj == null)
        {
            obj = new Singleton();
            return obj;
        }
        else
        {
            return obj;
        }
    }

    public void printSingleton()
    {
        Console.WriteLine("Singleton Class");
    }
}
